### PR TITLE
feat: scrub payloads with forbidden words

### PR DESCRIPTION
**Pull request by pauldambra**
Just like with headers we should be conservative in what we capture in payloads

If the body of the request (or response) contains any of the following we don't capture it

    'password',
    'secret',
    'passwd',
    'api_key',
    'apikey',
    'auth',
    'credentials',
    'mysql_pwd',
    'privatekey',
    'private_key',
    'tokenconst',
    
    If it matches SSN or CC regex we don't capture it
    
    You can add to this but not replace it

### CODE REVIEW

File: src/extensions/replay/config.ts [hunk 1]
```diff
@@ -114,17 +132,46 @@ const limitPayloadSize = (
...
+    _each(PAYLOAD_CONTENT_DENY_LIST, (text) => {
+        if (scrubbed?.length && scrubbed?.indexOf(text) !== -1) {
+            scrubbed = LOGGER_PREFIX + ' ' + label + ' body might contain: ' + text
+        }
+    })
+
+    return scrubbed
+}
+
+function scrubPayloads(capturedRequest: CapturedNetworkRequest | undefined): CapturedNetworkRequest | undefined {
```
Comment (marandaneto) [inline src/extensions/replay/config.ts:164]: Probably worth scrubbing the URL as well.
Eg google.com/login/password/123 or google.com/login?password=123 will be captured and there's a PII risk

File: src/extensions/replay/config.ts [hunk 2]
```diff
@@ -60,10 +64,24 @@ const HEADER_DENYLIST = [
...
+    'privatekey',
+    'private_key',
+    'tokenconst',
+]
+
 // we always remove headers on the deny list because we never want to capture this sensitive data
 const removeAuthorizationHeader = (data: CapturedNetworkRequest): CapturedNetworkRequest => {
-    Object.keys(data.requestHeaders ?? {}).forEach((header) => {
-        if (HEADER_DENYLIST.includes(header.toLowerCase())) delete data.requestHeaders?.[header]
+    _each(Object.keys(data.requestHeaders ?? {}), (header) => {
```
Comment (benjackwhite) [inline src/extensions/replay/config.ts:83]: Random one - I don't like the `_each` function - we should eventually make it explicit like `_iterateArray` and `_iterateObject` as currently it screws the typing

**Review by benjackwhite — APPROVED** [review]

File: src/extensions/replay/config.ts (hunk 1 above)
Comment (marandaneto) [reply src/extensions/replay/config.ts:164]: no need to block this PR but worth fixing as part of the TODOs (ps I was part of incidents with URL paths and query strings before), so definitely a must.

**Review by marandaneto — APPROVED** [review]

**Comment by pauldambra** [comment]
it's already maybe too keen since we now don't capture an API call that has `"has_social_auth": false` in the response

but, easier to relax after the fact 😊

**Pull request merged**